action_skill_analysis_cp4d.ipynb

```diff
@@ -0,0 +1,995 @@
+    "\n",
+    "2.1. [Obtain test data from Cloud Object Storage](#cos)<br>\n",
+    "2.2. [Evaluate the test data](#evaluate) <br>\n",
+    "2.3. [Analyze the test data](#testanalysis) <br>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2.1 Obtain test data from Cloud Object Storage<a id='cos'></a>\n",
+    "\n",
+    "Upload a test set in tsv format. Each line in the file should have only `User_Input<tab>Intent`  \n",
+    "\n",
+    "For example:\n",
+    "```\n",
+    "hello how are you<tab>Greeting  \n",
+    "I would like to talk to a human<tab>AgentHandoff  \n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "importlib.reload(skills_util)\n",
+    "#Separator: Use '\\t' for tab separated data, ',' for comma separated data\n",
+    "separator = \"\\t\"\n",
+    "\n",
+    "test_set_path = \"./test.tsv\"\n",
+    "test_df = skills_util.process_test_set(test_set_path, lang_util,separator)\n",
+    "\n",
+    "display(Markdown(\"### Random Test Sample\"))\n",
+    "display(HTML(test_df.sample(n=min(10, len(test_df))).to_html(index=False)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2.2 Evaluate the test data<a id='evaluate'></a>\n",
+    "These steps can take time if you have a large test set.  \n",
+    "\n",
+    "**<font color=red>Note</font>**: You will be charged for calls made from this notebook based on your Watson Assistant plan. The user_id will be the same for all message calls."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Change Assistant API version if needed\n",
+    "# Find Latest --> https://cloud.ibm.com/apidocs/assistant-v2#versioning\n",
+    "API_VERSION = '2021-11-27'\n",
+    "\n",
+    "# For ICP(IBM Cloud Private), you can disable SSL verification by changing this to True\n",
+    "DISABLE_SSL_VERTIFICATION = False \n",
+    "\n",
+    "conversation = skills_util.retrieve_conversation(username=username,\n",
+    "                                             password=password,                                                       \n",
+    "                                             authenticator_url=CP4D_URL,\n",
+    "                                             url=SERVICE_URL,\n",
+    "                                             api_version=API_VERSION,\n",
+    "                                             sdk_version=\"V2\", # DO NOT CHANGE THIS ARG\n",
+    "                                             cp4d_auth=True,\n",
+    "                                             )\n",
+    "\n",
+    "# if you have a BEARER_TOKEN, you can uncomment the code below and directly use your token\n",
+    "# conversation = skills_util.retrieve_conversation(bearer_token=BEARER_TOKEN)\n",
```

Comment:
it should be `skills_util.retrieve_conversation(bearer_token=BEARER_TOKEN, url=SERVICE_URL)`